patch-cli.md

```diff
@@ -0,0 +1,153 @@
+---
+layout: page
+title: Querying for a patch plan via CLI
+permalink: /patch-plan/
+parent: GUAC demos
+grand_parent: Getting started with GUAC
+nav_order: 3
+---
+
+# Query for a patch plan via CLI Demo
+
+In this demo, we will utilize a Go CLI that will allow us to input a vulnerable
+purl (package URL) and see what else is affected by this package being
+vulnerable-- meaning we are querying for the package's dependents. This tutorial
+assumes you know which package has some sort of vulnerability or malware
+affecting it, or which package is your concern.
+
+We will get back a patch plan which includes a visual representation of the
+blast radius, meaning a link to view the subgraph of dependent packages in the
+GUAC visualizer. It will also return a list of frontiers. These frontiers
+represent what can be patched to resolve the vulnerability, starting from the
+lowest level possible, i.e. the package itself.  
+If you don't have access to fix this package directly, you could look in the
+next frontier and so on. It will also list out and points of contact for a given
+node, if available in the graph.  
+This will help you know who to contact to resolve the issue. All these outputs
+together can be used to help figure out the best plan of action in order to
+resolve a security incident.
+
+## Step 1: Setup GUAC with Docker Compose
+
+Follow the [Docker Compose setup]({{ site.baseurl }}{%link setup.md %}).
+
+## Step 2: Clone GUAC
+
+1. Clone GUAC to a local directory:
+
+   ```bash
+   git clone https://github.com/guacsec/guac.git
+   ```
+
+2. Clone GUAC data (this is used as test data for this demo):
+
+   ```bash
+   git clone https://github.com/guacsec/guac-data.git
+   ```
+
+The rest of the demo will assume you are in the GUAC directory.
+
+```bash
+cd guac
+```
+
+## Step 3: Build the GUAC binaries
+
+Build the GUAC binaries using the `make` command:
+
+```bash
+make
+```
+
+## Step 4: Run the GUAC Visualizer
+
+To get the GUAC visualizer up and running, follow the [GUAC visualizer
+setup]({{ site.baseurl }}{%link guac-visualizer.md %}). This will be used in
+this demo to show the various paths from package to vulnerability.
+
+## Step 5: Run the Query Patch CLI
+
+In this demo, we will query for the patch plan of the following package (at the
+name level):
+
+```bash
+pkg:golang/github.com/antlr/antlr4/runtime/go/antlr
```

Comment:
do we need to ingest guac-data first or is that already taken care of?

Reply:
Its in the setup with docker compose I linked in step 1